lti_consumer/plugin/views.py

```diff
@@ -416,3 +425,163 @@ def scores(self, request, *args, **kwargs):  # pylint: disable=unused-argument
             status=status.HTTP_201_CREATED,
             headers=headers
         )
+
+
+class LtiNrpsContextMembershipViewSet(viewsets.ReadOnlyModelViewSet):
+    """
+    LTI NRPS Context Membership Service endpoint.
+    """
+
+    # Custom permission classes for LTI APIs
+    authentication_classes = [Lti1p3ApiAuthentication]
+    permission_classes = [LtiNrpsContextMembershipsPermissions]
+
+    # Renderer classes to accept LTI NRPS content types
+    renderer_classes = [
+        MembershipResultRenderer,
+    ]
+
+    # Custom pagination class to support pagination via Link Header
+    pagination_class = LinkHeaderPagination
+
+    @staticmethod
+    def map_lti_to_lms_role(lti_role=None):
+        """
+        Maps given lti role to corresponding edx role.
+        """
+        if lti_role:
+            for edx_role, context_roles in LTI_1P3_CONTEXT_ROLE_MAP.items():
+                if lti_role in context_roles:
+                    return edx_role
+
+        # if no NRPS filter role provided or there was no match in LTI_1P3_CONTEXT_ROLE_MAP
+        return None
```

Comment:
@shimulch This does not filter the queryset properly since it returns on the first match. 
See my output below:
```
{
  "id": "http://localhost:18000/api/lti_consumer/v1/lti/4/memberships?role=http%3A%2F%2Fpurl.imsglobal.org%2Fvocab%2Flis%2Fv2%2Fmembership",
  "context": {
    "id": "course-v1:OpenCraft+LTI202+2021"
  },
  "members": [
    {
      "status": "Active",
      "user_id": "a1752805-5281-437c-ae82-4f2430b3384c",
      "roles": [
        "http://purl.imsglobal.org/vocab/lis/v2/membership#Learner",
        "http://purl.imsglobal.org/vocab/lis/v2/membership#Administrator",
        "http://purl.imsglobal.org/vocab/lis/v2/membership#Instructor"
      ],
      "name": "full name test",
      "email": "edx@example.com"
    },
    {
      "status": "Active",
      "user_id": "bc73ed9f-10c5-4378-9ae9-14a57ed14592",
      "roles": [
        "http://purl.imsglobal.org/vocab/lis/v2/membership#Learner"
      ],
      "name": "",
      "email": "honor@example.com"
    }
  ]
}
```

The second entry should not have been returned in this case.
We need a new map from IMS to Open edX roles in the constants file.

Reply:
@giovannicimolin, the `role` filter arg is not correct. `http://purl.imsglobal.org/vocab/lis/v2/membership` is not a valid role. From the [spec](https://www.imsglobal.org/spec/lti/v1p3/#lis-vocabulary-for-context-roles), the role will be one -

```
http://purl.imsglobal.org/vocab/lis/v2/membership#Administrator
http://purl.imsglobal.org/vocab/lis/v2/membership#ContentDeveloper
http://purl.imsglobal.org/vocab/lis/v2/membership#Instructor
http://purl.imsglobal.org/vocab/lis/v2/membership#Learner
http://purl.imsglobal.org/vocab/lis/v2/membership#Mentor
```

But yes there is a bug, the API should not have returned any member at all.

Reply:
@giovannicimolin, I've pushed a fix for this bug according to my previous comment. Also, I suggest to urlencode the role URI first before submitting the `GET` request. The `#` part gets discarded if not urlencoded.